### PR TITLE
sql_database_instance disk_size and disk_autoresize can both be true 

### PR DESCRIPTION
**Pull request by pjo256**
Update sql_database_instance docs since disk_size and disk_autoresize are not mutually exclusive

Closes https://github.com/hashicorp/terraform-provider-google/issues/12618

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] ~~Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).~~
- [ ] ~~[Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).~~
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:note
sql: improved disk_size and disk_autoresize docs
```

### CODE REVIEW

File: mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown [hunk 1]
```diff
@@ -236,11 +236,11 @@ The `settings` block supports:
 
 * `collation` - (Optional) The name of server instance collation.
 
-* `disk_autoresize` - (Optional, Default: `true`) Enables auto-resizing of the storage size. Set to false if you want to set `disk_size`.
+* `disk_autoresize` - (Optional, Default: `true`) Enables auto-resizing of the storage size.
 
 * `disk_autoresize_limit` - (Optional, Default: `0`) The maximum size to which storage capacity can be automatically increased. The default value is 0, which specifies that there is no limit.
 
-* `disk_size` - (Optional, Default: `10`) The size of data disk, in GB. Size of a running instance cannot be reduced but can be increased. If you want to set this field, set `disk_autoresize` to false.
+* `disk_size` - (Optional, Default: `10`) The size of data disk, in GB. Size of a running instance cannot be reduced but can be increased.
```
Comment (ScottSuarez) [inline mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown:243]:
```suggestion
* `disk_size` - (Optional) The size of data disk, in GB. Size of a running instance cannot be reduced but can be increased.
```

I don't agree with including the api default here. It's not a terraform level default so I would omit it. You could mention in the description that the api sets a default value and link to corroborating api documentation.

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown (hunk 1 above)
Comment (pjo256) [reply mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown:243]: Should we remove the `disk_size` default note entirely? It appears to be an [undocumented default](https://github.com/GoogleCloudPlatform/magic-modules/pull/6596/files#diff-b34dfa25708e844d450cdb21f989baa23ffe466f2b817267606b10adc69048c1R307). I'll move the default into the description for other fields.

Comment (pjo256) [reply mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown:243]: Also, the default value for `connect_retry_interval` of 60s in our docs is mentioned as a MySQL-specific default in [PostgreSQL](https://cloud.google.com/sql/docs/postgres/admin-api/rest/v1beta4/instances#settings) and [SQL Server](https://cloud.google.com/sql/docs/sqlserver/admin-api/rest/v1beta4/instances#settings) 🤔

Comment (ScottSuarez) [reply mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown:243]: > mention that it's the minimum size instead

I think sounding off that it's an API enforced minimum is a good idea !

**Review by ScottSuarez — APPROVED** [review]
Thanks for the work !

**Pull request merged**